boutdata/tests/make_test_data.py

```diff
@@ -0,0 +1,390 @@
+
+    create3D_t("field3d_t_1")
+    create3D_t("field3d_t_2")
+
+    def create3D(name):
+        var = outputfile.createVariable(name, float, ("x", "y", "z"))
+
+        data = rng.random((localnx, localny, localnz))
+        var[:] = data
+        for key, value in expected_attributes[name].items():
+            var.setncattr(key, value)
+
+        result[name] = data[xslice, yslice, zslice]
+
+    create3D("field3d_1")
+    create3D("field3d_2")
+
+    # Field2D
+    def create2D_t(name):
+        var = outputfile.createVariable(name, float, ("t", "x", "y"))
+
+        data = rng.random((nt, localnx, localny))
+        var[:] = data
+        for key, value in expected_attributes[name].items():
+            var.setncattr(key, value)
+
+        result[name] = data[:, xslice, yslice]
+
+    create2D_t("field2d_t_1")
+    create2D_t("field2d_t_2")
+
+    def create2D(name):
+        var = outputfile.createVariable(name, float, ("x", "y"))
+
+        data = rng.random((localnx, localny))
+        var[:] = data
+        for key, value in expected_attributes[name].items():
+            var.setncattr(key, value)
+
+        result[name] = data[xslice, yslice]
+
+    create2D("field2d_1")
+    create2D("field2d_2")
+
+    # FieldPerp
+    def createPerp_t(name):
+        var = outputfile.createVariable(name, float, ("t", "x", "z"))
+
+        data = rng.random((nt, localnx, localnz))
+        var[:] = data
+        for key, value in expected_attributes[name].items():
+            var.setncattr(key, value)
+        var.setncattr("yindex_global", fieldperp_global_yind)
+
+        result[name] = data[:, xslice, zslice]
+
+    createPerp_t("fieldperp_t_1")
+    createPerp_t("fieldperp_t_2")
+
+    def createPerp(name):
+        var = outputfile.createVariable(name, float, ("x", "z"))
+
+        data = rng.random((localnx, localnz))
+        var[:] = data
+        for key, value in expected_attributes[name].items():
+            var.setncattr(key, value)
+        var.setncattr("yindex_global", fieldperp_global_yind)
+
+        result[name] = data[xslice, zslice]
+
+    createPerp("fieldperp_1")
+    createPerp("fieldperp_2")
+
+    # Time-dependent array
+    def createScalar_t(name):
+        var = outputfile.createVariable(name, float, ("t",))
+
+        data = rng.random(nt)
+        var[:] = data
+
+        result[name] = data
+
+    createScalar_t("t_array")
+    createScalar_t("scalar_t_1")
+    createScalar_t("scalar_t_2")
+
+    # Scalar
+    def createScalar(name, value):
+        var = outputfile.createVariable(name, type(value))
+
+        var[...] = value
+
+        result[name] = value
+
+    createScalar("BOUT_VERSION", 4.31)
+    for key, value in grid_info.items():
+        createScalar(key, value)
+    nxpe = grid_info["NXPE"]
+    createScalar("PE_XIND", i % nxpe)
+    createScalar("PE_YIND", i // nxpe)
+    createScalar("MYPE", i)
+
+    return result
+
+
+def concatenate_data(data_list, *, nxpe, fieldperp_yproc_ind):
```

Comment:
Would a short docstring be helpful here?
Is it for example correct that `data_list` is a list where the index refers to the processor number and the value is a dict on the `expected_attrbutes` form?